raiden_libs/messages/message.py

```diff
@@ -35,45 +33,20 @@ def type(self, value):
 
     def serialize_full(self, private_key):
         """Serialize message to a standardized format, including message envelope"""
-
-        msg_data = self.assemble_message()
-        msg_data['header']['sender'] = private_key_to_address(private_key)
-        msg_data = json.dumps(msg_data)
-        return json.dumps({
-            'signature': self.sign_data(private_key, msg_data),
-            'data': msg_data
-        })
+        msg = self.serialize_data()
+        msg['message_type'] = self._type
+        return json.dumps(msg)
 
     def sign_data(self, private_key, data: str):
         data_hash = keccak256(data)
         return encode_hex(sign(private_key, data_hash))
 
-    def assemble_message(self):
-        assert isinstance(self.type, str)
-        return {
-            'header': {
-                'type': self.type,
-                'timestamp': time.time(),
-                'sender': None
-            },
-            'body': self.serialize_data()
-        }
-
     @staticmethod
     def deserialize(data):
         if isinstance(data, str):
             json_message = json.loads(data)
         else:
             json_message = data
         jsonschema.validate(json_message, Message.json_schema)
-        json_data = json.loads(json_message['data'])
-        cls = deserialize(json_data)
-        cls.header = json_data['header']
-        if len(json_message['signature']) != 132:
-            raise MessageFormatError('Invalid signature value')
-        cls.signer = to_checksum_address(
-            eth_verify(decode_hex(json_message['signature']), json_message['data'])
-        )
-        if cls.signer != json_data['header']['sender']:
-            raise MessageSignatureError('Signature does not match the sender!')
+        cls = deserialize(json_message)
```

Comment:
I wouldn't call this `cls`. This is the deserialized message.

Reply:
it's _instance_ of the message, thus _cls_, like when using `@classmethod` decorator